### PR TITLE
Render in tiles' titles.

### PR DESCRIPTION
**Pull request by rpiaggio**
By using React's portals, tiles contents can now render things into the tile's title.

`Tile` now accepts a `render` function `Tile.RenderInTitle ~=> VdomNode` instead of children. The contents renderer therefore now receive a `Tile.RenderInTitle` which is just a `VdomNode ~=> VdomNode` that when called, will create a portal into a `<.span` in the title.

Undo buttons have been migrated to use this mechanism.

### CODE REVIEW

File: common/src/main/scala/explore/components/Tile.scala [hunk 1]
```diff
@@ -45,11 +53,16 @@ object Tile {
          (1024                           -> TileLGW)
     )
 
+  def renderPortal(mountNode: Option[raw.ReactDOM.Container], info: VdomNode): VdomNode =
+    mountNode.map(node => ReactPortal(info, node))
```
Comment (cquiroz) [inline common/src/main/scala/explore/components/Tile.scala:57]: Is ReactPortak defined in scalajs-react?

**Review by cquiroz — APPROVED** [review]
Extraordinary, this is going to be very useful

**Review by toddburnside — APPROVED** [review]
Very cool!

File: common/src/main/scala/explore/components/Tile.scala (hunk 1 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/components/Tile.scala:57]: Yes

**Pull request merged**